source/vistas/ui/windows/linreg_dialog.py

```diff
@@ -11,54 +11,374 @@
+        #DEPENDENT VARIABLE
+        ctl_sizer.Add(wx.StaticText(self.panel, -1, 'Dependent variable:'), flag=wx.TOP|wx.LEFT|wx.RIGHT, border=12)
+
         self.dv_chooser = wx.Choice(self.panel, choices=['-'] + data_choices)
         ctl_sizer.Add(self.dv_chooser, flag=wx.LEFT|wx.RIGHT, border=12)
 
+        #PLOT TYPE
         ctl_sizer.Add(wx.StaticText(self.panel, -1, 'Plot type:'), flag=wx.TOP|wx.LEFT|wx.RIGHT, border=12)
+
         self.plot_type = wx.RadioBox(self.panel, choices=['scatterplot', 'heatmap'])
         ctl_sizer.Add(self.plot_type, flag=wx.LEFT|wx.RIGHT, border=10)
+
+        #AXIS TYPE
         ctl_sizer.Add(wx.StaticText(self.panel, -1, 'Axis type:'), flag=wx.TOP|wx.LEFT|wx.RIGHT, border=12)
-        self.axis_type = wx.RadioBox(self.panel, choices=['fixed', 'adaptive'])
+
+        self.axis_type = wx.RadioBox(self.panel, choices=['Fit All', 'Adaptive', 'Zoom'])
         ctl_sizer.Add(self.axis_type, flag=wx.LEFT|wx.RIGHT, border=10)
+
         self.Bind(wx.EVT_RADIOBOX, self.doPlot)
 
+        #BLANK SPACER
+        ctl_sizer.Add(wx.StaticText(self.panel, -1, ''), flag=wx.TOP|wx.EXPAND, border=200)
+
+
+        #ZOOM CONTROLS
+
+        #Zoom button for dragging a box
+        self.zoom_box = wx.Button(self.panel, label="Box")
+        zoom_sizer.Add(self.zoom_box)
+
+        self.zoom_box.Bind(wx.EVT_BUTTON, self.zoom_mode_change)
+
+        #ZOOM SLIDER
+        self.zoom = wx.Slider(self.panel, value = 0, minValue = 0, maxValue = 49, size = (600,-1), style = wx.SL_HORIZONTAL)
+        zoom_sizer.Add(self.zoom, flag=wx.LEFT, border = 10)
+
+        self.zoom.Bind(wx.EVT_SCROLL, self.disableZoom)
+        self.zoom.Bind(wx.EVT_SCROLL, self.doPlot)
+
+        #Graph canvas
         self.fig = mpl.figure.Figure()
         self.canvas = wxagg.FigureCanvasWxAgg(self.panel, -1, self.fig)
-        top_sizer.Add(self.canvas, 1, wx.EXPAND)
-        self.Bind(wx.EVT_LISTBOX, self.doPlot)
-        self.Bind(wx.EVT_CHOICE, self.doPlot)
+
+        self.right_sizer.Add(self.canvas, 1, wx.EXPAND)
+        self.right_sizer.Add(zoom_sizer, flag=wx.EXPAND|wx.TOP|wx.LEFT|wx.RIGHT, border = 10)
+        top_sizer.Add(self.right_sizer)
+
+        #EVENTS
+
+        self.Bind(wx.EVT_LISTBOX, self.resetGraph)
+        self.Bind(wx.EVT_CHOICE, self.resetGraph)
+
         self.Bind(wx.EVT_CLOSE, self.onClose)
+
+        #Move along with timeline
         get_main_window().Bind(EVT_TIMELINE_CHANGED, self.doPlot)
+
+        #Mouse events on graph (Matplotlib events)
+        self.fig.canvas.mpl_connect('button_press_event', self.mouse_click)
+        self.fig.canvas.mpl_connect('button_release_event', self.mouse_up)
+        self.fig.canvas.mpl_connect('motion_notify_event', self.mouse_move)
+
+        #Open in the center of VISTAS main window
         self.CenterOnParent()
         self.panel.Layout()
         self.Show()
 
+    #Resets zoom controls so that graph can be made with new bounds
+    def resetGraph(self, event):
```

Comment:
Function names in Python should be lower case, with underscores `_` to separate words. E.g., `reset_graph`. wxPython will throw you off here. Since it's a port of a C++ library, it stick to the naming convention of the original library.

https://www.python.org/dev/peps/pep-0008/#id34